clients/include/testing_local_matrix.hpp

```diff
@@ -715,4 +715,2109 @@ bool testing_local_matrix_set_data_ptr(Arguments argus)
+    //
+    //for(int64_t i = 0; i < nnz; ++i)
+    //{
+    //    EXPECT_EQ(actual_values[i], expected_values[i]);
+    //}
+    //
+    //delete[] row_offsets;
+    //delete[] col_indices;
+    //delete[] values;
+    //delete[] expected_values;
+    //delete[] actual_values;
+}
+
+template <typename T>
+void testing_add_scalar(Arguments argus)
+{
+    LocalMatrix<T> matrix;
+    getTestMatrix<T>(argus, matrix);
+
+    int64_t nnz    = matrix.GetNnz();
+    T*      values = new T[nnz];
+    getMatrixVal(matrix, values);
+
+    // Perform AddScalar
+    matrix.AddScalar(1.0);
+
+    // Validate the result
+    T* values_new = new T[nnz];
+    getMatrixVal(matrix, values_new);
+    for(int64_t i = 0; i < nnz; ++i)
+    {
+        EXPECT_NEAR(values_new[i], values[i] + 1.0, getTolerance<T>() * std::abs(values_new[i]));
+    }
+    delete[] values;
+    delete[] values_new;
+}
+
+template <typename T>
+void testing_replace_column_vector(Arguments argus)
+{
+    LocalMatrix<T> matrix;
+    getTestMatrix<T>(argus, matrix);
+
+    int n = matrix.GetN();
+    int m = matrix.GetM();
+
+    // Test replacing each column with a known vector and verify
+    for(int col = 0; col < n; ++col)
+    {
+        LocalVector<T> vec;
+        vec.Allocate("ColumnVector", m);
+        for(int i = 0; i < m; ++i)
+            vec[i] = static_cast<T>(i + 100 * col); // Unique values per column
+
+        // Replace column col
+        EXPECT_NO_THROW(matrix.ReplaceColumnVector(col, vec));
+
+        // Extract column col
+        LocalVector<T> extracted_vec;
+        extracted_vec.Allocate("ExtractedColumn", m);
+        EXPECT_NO_THROW(matrix.ExtractColumnVector(col, &extracted_vec));
+
+        // Verify the replacement
+        for(int i = 0; i < m; ++i)
+            EXPECT_EQ(extracted_vec[i], vec[i]);
+    }
+}
+
+template <typename T>
+void testing_extract_column_vector(Arguments argus)
+{
+    LocalMatrix<T> matrix;
+    getTestMatrix<T>(argus, matrix);
+
+    int n = matrix.GetN();
+    int m = matrix.GetM();
+
+    // Extract each column and compare with dense representation
+    auto dense = extract_dense_matrix(matrix);
+
+    for(int col = 0; col < n; ++col)
+    {
+        LocalVector<T> vec;
+        vec.Allocate("ExtractedColumn", m);
+
+        // Extract column vector
+        EXPECT_NO_THROW(matrix.ExtractColumnVector(col, &vec));
+
+        // Validate the result
+        for(int row = 0; row < m; ++row)
+        {
+            EXPECT_EQ(vec[row], dense[row][col]);
+        }
+    }
+}
+
+template <typename T>
+void testing_set_data_ptr_DIA()
+{
+    // Create a LocalMatrix
+    LocalMatrix<T> mat;
+
+    // Define DIA matrix parameters
+    int64_t nnz      = 12; // Number of non-zero elements
+    int64_t nrow     = 6; // Number of rows
+    int64_t ncol     = 6; // Number of columns
+    int     num_diag = 2; // Number of diagonals
+
+    // Allocate memory for DIA data
+    int* offset = new int[num_diag]{0, 1}; // Diagonal offsets
```

Comment:
Where is the `delete []`?

Reply:
After `SetDataPtrDIA` the pointers are null pointers.